presto-spark-base/src/main/java/com/facebook/presto/spark/execution/http/PrestoSparkHttpWorkerClient.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spark.execution.http;
+
+import com.facebook.airlift.http.client.HttpClient;
+import com.facebook.airlift.http.client.HttpStatus;
+import com.facebook.airlift.http.client.Request;
+import com.facebook.airlift.http.client.Response;
+import com.facebook.airlift.http.client.ResponseHandler;
+import com.facebook.airlift.log.Logger;
+import com.facebook.presto.execution.TaskId;
+import com.google.common.util.concurrent.ListenableFuture;
+import io.airlift.units.DataSize;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+import java.net.URI;
+
+import static com.facebook.airlift.http.client.HttpStatus.familyForStatusCode;
+import static com.facebook.airlift.http.client.HttpUriBuilder.uriBuilderFrom;
+import static com.facebook.airlift.http.client.Request.Builder.prepareDelete;
+import static com.facebook.airlift.http.client.Request.Builder.prepareGet;
+import static com.facebook.airlift.http.client.StatusResponseHandler.createStatusResponseHandler;
+import static com.facebook.presto.client.PrestoHeaders.PRESTO_MAX_SIZE;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * An abstraction of HTTP client that communicates with the locally running Presto worker process. It exposes worker endpoints to simple method calls.
+ */
+@ThreadSafe
+public class PrestoSparkHttpWorkerClient
+{
+    private static final Logger log = Logger.get(PrestoSparkHttpWorkerClient.class);
+
+    private final HttpClient httpClient;
+    private final URI location;
+
+    public PrestoSparkHttpWorkerClient(HttpClient httpClient, URI location)
```

Comment:
- Is it possible to pass taskId here instead of using it through all APIs? Because taskId will be fixed once the native sidecar is launched. Then at the time of creating this client, we should have already known about the id.
- Then we can remove taskId in all interfaces. We can also implement this class with `RpcShuffleClient` interface.